sound/soc/intel/boards/sof_es8336.c

```diff
@@ -152,14 +159,37 @@ static void sof_es8316_exit(struct snd_soc_pcm_runtime *rtd)
 	snd_soc_component_set_jack(component, NULL, NULL);
 }
 
+static int sof_es8336_quirk_cb(const struct dmi_system_id *id)
+{
+	quirk = (unsigned long)id->driver_data;
+	return 1;
+}
+
+static int sof_es8336_quirk_tgl_cb(const struct dmi_system_id *id)
+{
+	quirk = (unsigned long)id->driver_data;
+	gpio_mapping = quirk_acpi_es8336_gpios;
+
+	return 1;
+}
+
 static const struct dmi_system_id sof_es8336_quirk_table[] = {
 	{
+		.callback = sof_es8336_quirk_cb,
 		.matches = {
 			DMI_MATCH(DMI_SYS_VENDOR, "CHUWI Innovation And Technology"),
 			DMI_MATCH(DMI_BOARD_NAME, "Hi10 X"),
 		},
 		.driver_data = (void *)SOF_ES8336_SSP_CODEC(2)
 	},
+	{
+		.callback = sof_es8336_quirk_tgl_cb,
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "IP3 tech"),
+			DMI_MATCH(DMI_BOARD_NAME, "WN1"),
+		},
+		.driver_data = (void *)SOF_ES8336_SSP_CODEC(0)
```

Comment:
I think it would be better to use a new quirk here, e.g. SOF_ES886_TGL_GPIO_QUIRK, that way you can initialize the gpio_mapping in a single callback. e.g.

````c
if (quirk & SOF_ES886_TGL_GPIO_QUIRK) {
    gpio_mapping = quirk_acpi_es8336_gpios;
}
````

Reply:
Hi Pierre, 
Updated it according to your comment, please help to check the latest code, thanks.